test/node/error_test.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const BSON = require('../register-bson');
+const BSONTypeError = BSON.BSONTypeError;
+const BSONError = BSON.BSONError;
+
+describe('BSON error tests', function () {
```

Comment:
Can we restructure these just a bit for clarity? I think if we add a separate describe block for `BSONError` and `BSONTypeError` and then put the corresponding `it` statements instead each, it would set up a better precedent in case we decide to add more errors. The `it` statements would be something like `should evaluate true on instanceof BSONError and Error`/`should evaluate true on instanceof BSONTypeError and TypeError`, and then for the assertions, I would check both `instanceof ... to.be.true` and `to.be.instanceOf(...)` because that way we ensure that our tests work as expected and that the actual `instanceof` check that an external user might use would work as expected even if mocha decides to change its underlying implementation of `instanceOf` for some reason; and then for the second `it`, make it a `should correctly set the name and message properties`